### ISSUE TITLE
Calling `fail()` does not cause `process.exitCode = 1` without `--verbose`.

### ISSUE THREAD

**Issue by dtinth** [danger/danger-js#706]
When there is a failure, Danger says `Failing the build, there is 1 fail.` but the `danger` process completes successfully.

I looked at the code, and found that `process.exitCode = 1` is in `Executor#handleResultsPostingToSTDOUT`, which is only executed when `--verbose` is set. Therefore, if `--verbose` flag is not set, the build will pass despite Danger said it would fail.

In my opinion, this violates the principle of least surprise in 2 ways: 1) Why does the build not fail despite Danger saying it world, and 2) Why passing `--verbose` affects the exit code of Danger.

**Comment by orta** [comment]
It failed your build via the GitHub status API - Oddly enough, most people felt that doing it this way is the least surprise version of Danger (and what has been the default for about 4 years)  which I think I'm pretty happy with.

In the ruby version someone added a flag for always giving a failing process code on fail, but no-one has ported it over to Danger JS. Happy to have that in 👍

**Comment by dtinth** [comment]
I see… 👍 Thanks for the insights.

For our use cases, we want to have failures actually fail the CI build because we’re using CircleCI Workflows. We use Danger to check if a PR is properly labelled and categorized. That means the problem we try to detect using Danger is not code-related (and can be fixed without committing any code to the repository).

However, CircleCI workflow only allows you to rebuild jobs only when it fails. In this default scenario, Danger failed but exits with code 0. Therefore, the CI job passed. After I fix the problem, I only have the option rerun the entire workflow, causing all tests to be run again.

![image](https://user-images.githubusercontent.com/193136/47441238-e4b8ee00-d7d9-11e8-9ee8-8c7e299f20a2.png)

Here, I added `--verbose` to make Danger also crash the CI job. Now CircleCI offers the option to rerun just the failed jobs. This saves 3 minutes of build time.

![image](https://user-images.githubusercontent.com/193136/47441326-1a5dd700-d7da-11e8-9895-1b6afebe3cde.png)

Since we worked around this issue using the `--verbose` flag, I don’t have an immediate need to fix this. But I agree about creating a dedicated flag that causes `process.exitCode = 1` on failure. Maybe put this issue on [Hacktoberfest](https://hacktoberfest.digitalocean.com/)? 😉
